### ISSUE TITLE
Language change is not detected by NVDA

### ISSUE THREAD

**Issue by RubyDo** [nvaccess/nvda#12611]
### Steps to reproduce: 
1) In a page with language declaration in English,  use code inspector and put language attribute to identify changes in the human language parts (image, div, link) to French. 
`<img src="xxx" lang="fr" alt="Page en Français"/>`
`<a href="xxx" lang="fr">Page en Français"</a>`
`<div lang="fr">xxx</div>`

2) Open NVDA and read the elements

3) Notice NVDA didn't identify the change in language from English to French. 
More on Language of Parts on WCAG [H58 : Using language attributes to identify changes in the human language](https://www.w3.org/WAI/WCAG21/Techniques/html/H58.html)

### Actual behavior: 
NVDA reads the parts in English 

### Expected behavior: 
NVDA should read in French

#### NVDA installed/portable/running from source:
NVDA installed on laptop

#### NVDA version: 
NVDA 2020.4 

#### Windows version: 
Windows 10 Home version 21H1

#### Name and version of other software in use when reproducing the issue: 
NVDA 2020.4 with Chrome v 91.0.4472.124 

#### Does the issue still occur after restarting your computer? 
Yes

**Comment by seanbudd** [comment]
Hi @RubyDo,

Does this issue occur when visiting a page that has this HTML, **without** editing the source code? If so, can you create a [codepen ](https://codepen.io/pen/) example? 

NVDA stores a buffer to process accessibility information, which might not be updated when editing the source code.
I can't reproduce this otherwise.

**Comment by zersiax** [comment]
@RubyDo  can you try this on another type of element, e.g. a paragraph? I'm wondering if the lang attribute might just not work on alt texts either because of an oversight in NVDA or the fact that alt text just doesn't conform to a lang tag on its parent image given alt is an attribute, not the contents of a tag

**Comment by roberthwhitaker** [comment]
I'm experiencing this issue also on a test page I created: https://leftalign.cc/te-reo-maori-test.html

Items labelled with lang="mi" (the second set of examples) are not being read using that voice.

This also fails in the other direction. If Māori is set as the language the page is read using that voice, even though it's document level markup identifies it as English.

**Comment by RubyDo** [comment]
Hello @roberthwhitaker @zersiax zersiax
Apparently the language identification does not happen on the page level or on language switch 
Here is a demo on youtube [https://youtu.be/-HnqDhc72Lo](url)

Test details: 
-MS Edge, Firefox, Chrome: In a page with lang="en" declaration, I tested language switch to french in h1, p, span, a and none work. 
-Firefox: In a page with lang declaration to french lang="fr", none is read in french.  [https://cenw-wscoe.github.io/sgdc-cdts/docs/index-fr.html](url)

Here are the settings and versions used 
NVDA 2020.4, Windows 10 Home version 21H1
- Chrome Version 94.0.4606.61 (Official Build) (64-bit)
-Firefox 91.0.2 (64-bit)
- Microsoft Edge Version 94.0.992.31 (Official build) (64-bit)

CC @szet0018

**Comment by emmamcarthur** [comment]
@seanbudd any plans for a fix?

**Comment by MattiJarvinen** [comment]
This is still broken.

Related #12794
